spec.emu

```diff
@@ -0,0 +1,1319 @@
+        <h1>
+          ValidateBase64VLQGroupings (
+            _groupings_: a String,
+          ): ~unused~
+        </h1>
+        <dl class="header"></dl>
+        <emu-alg>
+          1. If _groupings_ contains any code unit other than:
+              <ul>
+                <li>U+002B (`+`), U+002C (`,`), U+002F (`/`), or U+003B (`;`);</li>
+                <li>U+0030 (`0`) to U+0039 (`9`);</li>
+                <li>U+0041 (`A`) to U+005A (`Z`);</li>
+                <li>U+0061 (`a`) to U+007A (`z`)</li>
+              </ul>
+              throw an error.
+          1. Return ~unused~.
+        </emu-alg>
+
+        <emu-note>These are the valid <emu-xref href="#sec-references-informative">base64</emu-xref> characters (excluding the padding character `=`), together with `,` and `;`.</emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-DecodeBase64VLQ" type="abstract operation">
+        <h1>
+          DecodeBase64VLQ (
+            _segment_: a String,
+            _position_: a Record with a non-negative integer [[Value]] field,
+          ): an integer or *null*
+        </h1>
+        <dl class="header"></dl>
+        <emu-alg>
+          1. Let _segmentLen_ be the length of _segment_.
+          1. If _position_.[[Value]] = _segmentLen_, return *null*.
+          1. Let _first_ be ConsumeBase64ValueAt(_segment_, _position_).
+          1. Assert: _first_ &lt; 64.
+          1. If _first_ modulo 2 is 0, let _sign_ be 1.
+          1. Else, let _sign_ be -1.
+          1. Let _value_ be min(floor(_value_ / 2), 2<sup>4</sup> - 1).
+          1. Let _nextShift_ be 16.
+          1. Let _currentByte_ be _first_.
+          1. Repeat, while floor(_currentByte_ / 2<sup>5</sup>) modulo 2 = 1,
+            1. If _position_.[[Value]] = _segmentLen_, throw an error.
+            1. Set _currentByte_ to ConsumeBase64ValueAt(_segment_, _position_).
+            1. Let _chunk_ be min(_value_, 2<sup>5</sup> - 1).
+            1. Set _value_ to _value_ + _chunk_ × _nextShift_.
+            1. If _value_ &ge; 2<sup>32</sup>, throw an error.
+            1. Set _nextShift_ to _nextShift_ × 2<sup>5</sup>.
+          1. If _value_ is 0 and _sign_ is -1, return -2<sup>31</sup>.
+          1. Return _sign_ × _value_.
+        </emu-alg>
+
+        <emu-clause id="ConsumeBase64ValueAt" type="abstract operation">
+          <h1>
+            ConsumeBase64ValueAt (
+              _string_: a String,
+              _position_: a Record with a non-negative integer [[Value]] field,
+            ): a non-negative integer
+          </h1>
+          <dl class="header"></dl>
+          <emu-alg>
+            1. Assert: _position_.[[Value]] is a non-negative integer smaller than the length of _string_.
+            1. Let _char_ be the substring of _string_ from _position_ to _position_+ 1.
+            1. Assert: _char_ is a valid <emu-xref href="#sec-references-informative">base64</emu-xref> character as defined by IETF RFC 4648.
```

Comment:
Should there be a step after this one that does `Set _position_.[[Value]] to _position_.[[Value]] + 1`?